Hot fix

bergano65 opened the pull request:
- http to https automatic redirect
- empty development team in  configuration proceeded right
- if application has no access to subscription owner get mail instead of empty

ischrei commented on CogsMinimizer/CogsMinimizer.csproj:
```diff
@@ -18,7 +18,7 @@
     <MvcBuildViews>false</MvcBuildViews>
     <UseIISExpress>true</UseIISExpress>
-    <IISExpressSSLPort />
+    <IISExpressSSLPort>44335</IISExpressSSLPort>
```
This is for debugging purposes. If you intended to do it, ok, otherwise please revert

ischrei commented on CogsMinimizer/Global.asax.cs:
```diff
@@ -21,5 +24,36 @@ protected void Application_Start()
+
+        protected void Application_BeginRequest(object sender, EventArgs args)
+        {
+            if (Context.Request.Url.Scheme == "https")
```
compare with ignore case (not sure if it can also gets "HTTPS" if the client sends it..

ischrei commented on CogsMinimizer/Global.asax.cs:
```diff
@@ -21,5 +24,36 @@ protected void Application_Start()
+            }
+            else
+            {
+                uri.Port = -1;
```
why? it seems wrong..

ischrei commented on CogsMinimizer/Global.asax.cs:
```diff
@@ -21,5 +24,36 @@ protected void Application_Start()
+                return;
+            }
+
+            if (_prevUri != null && _prevUri.AbsoluteUri == Context.Request.Url.AbsoluteUri)
```
Sorry but I don't understand it. Why do you need to previous URI? what if the previous URI was for another client request?

Anyhow all you need to do is redirect whenever you get a "http" request. I agree you can skip for localhost although it is better to simulate prod in local as much as possible.

ischrei commented on OfflineSubscriptionManager/App.config:
```diff
@@ -7,8 +7,8 @@
+    <add name="AzureWebJobsDashboard" connectionString="DefaultEndpointsProtocol=https;AccountName=evitensubminlog;AccountKey=Y+DFscPR1AOYAsPnAjKHJB634P68vNDjSuXkSrOyfEKbSmHEoEA54MxnGktL/I7oJ2/RZULskM4Ttgk91vABzg==;EndpointSuffix=core.windows.net"/>
+    <add name="AzureWebJobsStorage" connectionString="DefaultEndpointsProtocol=https;AccountName=evitensubminlog;AccountKey=Y+DFscPR1AOYAsPnAjKHJB634P68vNDjSuXkSrOyfEKbSmHEoEA54MxnGktL/I7oJ2/RZULskM4Ttgk91vABzg==;EndpointSuffix=core.windows.net"/>
     <add name="DefaultConnection" connectionString="Data Source=MAXIMSH-2010\SQLEXPRESS;Initial Catalog=DataAccess;Integrated Security=True;Connect Timeout=15;Encrypt=False;TrustServerCertificate=True;ApplicationIntent=ReadWrite;MultiSubnetFailover=False" providerName="System.Data.SqlClient"/>
   </connectionStrings>
```
please don't push such secrets.
As I recommended you, it is better to put all you secrets in key vault. this way you can have you own private environment without the need to replace those secrets.

Also I\m not sure who can see this repo, but you may consider regenerating your storage key now after it became available in GitHub

ischrei commented on OfflineSubscriptionManager/App.config:
```diff
@@ -46,6 +46,8 @@
     <add key="ida:ARMAuthorizationPermissionsAPIVersion" value="2014-07-01-preview" />
     <add key="ida:RequiredARMRoleOnSubscription" value="Contributor" />
 
+    <add key="DevTeam" value="eviten@microsoft.com" />
```
Also not sure you want to publish you email address this way. you call. anyhow please remove it from this file. you can have you own private configuration

ischrei commented on OfflineSubscriptionManager/EmailUtils.cs:
```diff
@@ -46,6 +46,13 @@ Subminimizer report
+            if (!analysisResult.IsSubscriptionAccessible)
+            {
+                message += $"<h3>Subscription isn't available for application scanning. Please mail to eviten@microsoft.com for instructions how to enable access to your subscription for SubMinimizer application</h3>";
+                message += "</body></html>";
```
put our DL instead, if you put it in the config file, take the value from there

ischrei commented on OfflineSubscriptionManager/Functions.cs:
```diff
@@ -94,7 +94,11 @@ private static void ReportSubscriptionAnalysisResult(SubscriptionAnalysisResult
+            string devTeam = ConfigurationManager.AppSettings["DevTeam"];
+            if (devTeam != null)
+            {
+                bcc.AddRange(devTeam.Split(';').Select(x => new Email(x)));
```
add our DL instead

bergano65 commented:
Development team is hot support  accounts. Only me think . even you don’t have be there. Configured there people receive copy of each mail sent. Why somebody should be in this group in addition to me?

From: ischrei [mailto:notifications@github.com]
Sent: Sunday, December 31, 2017 11:56 AM
To: shklar/SubMinimizer <SubMinimizer@noreply.github.com>
Cc: Evgeny Vitenberg <eviten@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [shklar/SubMinimizer] Hot fix (#77)


@ischrei commented on this pull request.

________________________________

In OfflineSubscriptionManager/Functions.cs<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23discussion_r159137073&data=02%7C01%7Ceviten%40microsoft.com%7Cbf96b7d660104f682f3e08d55034a1e7%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503109344597507&sdata=AX8Qii6KvfgBSJVF7J5tUZ8trsL7lwu7e4X1u3HVAwU%3D&reserved=0>:

> @@ -94,7 +94,11 @@ private static void ReportSubscriptionAnalysisResult(SubscriptionAnalysisResult

             }



             //Add BCC recepients - dev team, as configured in the app config

-            bcc.AddRange(ConfigurationManager.AppSettings["DevTeam"].Split(';').Select(x=> new Email(x)));

+            string devTeam = ConfigurationManager.AppSettings["DevTeam"];

+            if (devTeam != null)

+            {

+                bcc.AddRange(devTeam.Split(';').Select(x => new Email(x)));

add our DL instead

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23pullrequestreview-86044053&data=02%7C01%7Ceviten%40microsoft.com%7Cbf96b7d660104f682f3e08d55034a1e7%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503109344597507&sdata=3ZVWtVNtQUEXU4TaeCj8qrG%2FqwHeZubFRn46WH%2BnSPY%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FARCVLCsPGsn18BnNijGeh9K4vuQZKzleks5tF1oUgaJpZM4RPVhW&data=02%7C01%7Ceviten%40microsoft.com%7Cbf96b7d660104f682f3e08d55034a1e7%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503109344597507&sdata=MT3r9F8%2BETxVSPqhEbQJ7snwaz%2FCK3yQkuDXtyfOUn8%3D&reserved=0>.

bergano65 commented:
Since it’s question of configuring subscription for access of application available.
Why somebody should get this mail not only me?


From: ischrei [mailto:notifications@github.com]
Sent: Sunday, December 31, 2017 11:55 AM
To: shklar/SubMinimizer <SubMinimizer@noreply.github.com>
Cc: Evgeny Vitenberg <eviten@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [shklar/SubMinimizer] Hot fix (#77)


@ischrei commented on this pull request.

________________________________

In OfflineSubscriptionManager/EmailUtils.cs<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23discussion_r159137063&data=02%7C01%7Ceviten%40microsoft.com%7C6f704ad72f6949c523fc08d550349513%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503109124533328&sdata=1%2FfHa%2FumJ5tLRbifnq2jSqlkTKTu2hN5PeUVvqWio2w%3D&reserved=0>:

> @@ -46,6 +46,13 @@ Subminimizer report

             message += $"<H2>Analysis Date : {GetShortDate(sub.LastAnalysisDate)}</H2>";

             message += "<br>";



+            if (!analysisResult.IsSubscriptionAccessible)

+            {

+                message += $"<h3>Subscription isn't available for application scanning. Please mail to eviten@microsoft.com<mailto:eviten@microsoft.com> for instructions how to enable access to your subscription for SubMinimizer application</h3>";

+                message += "</body></html>";

put our DL instead, if you put it in the config file, take the value from there

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23pullrequestreview-86044043&data=02%7C01%7Ceviten%40microsoft.com%7C6f704ad72f6949c523fc08d550349513%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503109124533328&sdata=EuPycPTC4ld7jOk0963wnR77x1indqS8mo8APAHxwBQ%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FARCVLPXUS4kijgtT4Jhze1aWzR2VBMQtks5tF1n-gaJpZM4RPVhW&data=02%7C01%7Ceviten%40microsoft.com%7C6f704ad72f6949c523fc08d550349513%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503109124533328&sdata=bxY4UkiZ4kCB5gkyl8PSciln69mi1cApnQGK25bY4IE%3D&reserved=0>.

bergano65 commented:
OK, agree

From: ischrei [mailto:notifications@github.com]
Sent: Sunday, December 31, 2017 11:54 AM
To: shklar/SubMinimizer <SubMinimizer@noreply.github.com>
Cc: Evgeny Vitenberg <eviten@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [shklar/SubMinimizer] Hot fix (#77)


@ischrei commented on this pull request.

________________________________

In OfflineSubscriptionManager/App.config<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23discussion_r159137049&data=02%7C01%7Ceviten%40microsoft.com%7Ce0a17750c21b49ceaa4e08d5503473a0%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503108562792486&sdata=eNakVuxWgYPiRrhoUqMhwVRpdv8mrHj66d%2BAt%2FoYakk%3D&reserved=0>:

> @@ -46,6 +46,8 @@

     <add key="ida:ARMAuthorizationPermissionsAPIVersion" value="2014-07-01-preview" />

     <add key="ida:RequiredARMRoleOnSubscription" value="Contributor" />



+    <add key="DevTeam" value="eviten@microsoft.com<mailto:eviten@microsoft.com>" />

Also not sure you want to publish you email address this way. you call. anyhow please remove it from this file. you can have you own private configuration

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23pullrequestreview-86044034&data=02%7C01%7Ceviten%40microsoft.com%7Ce0a17750c21b49ceaa4e08d5503473a0%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503108562792486&sdata=nM829FtBkcgd%2Bbay3vN32Frc9t%2B8q98MzdPGQbVcPrs%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FARCVLGLlBC_9NFIaYzdeZH96C6WE5aT3ks5tF1nGgaJpZM4RPVhW&data=02%7C01%7Ceviten%40microsoft.com%7Ce0a17750c21b49ceaa4e08d5503473a0%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503108562792486&sdata=tuH8YTrp3U06yAM7rVbqtCY%2Baw2GIUtm33f1lUoRcbQ%3D&reserved=0>.

bergano65 commented:
Agree, will  regenerate secrets an remove from submit.

From: ischrei [mailto:notifications@github.com]
Sent: Sunday, December 31, 2017 11:53 AM
To: shklar/SubMinimizer <SubMinimizer@noreply.github.com>
Cc: Evgeny Vitenberg <eviten@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [shklar/SubMinimizer] Hot fix (#77)


@ischrei commented on this pull request.

________________________________

In OfflineSubscriptionManager/App.config<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23discussion_r159137024&data=02%7C01%7Ceviten%40microsoft.com%7Ccd85e1fe4a20401c2dbf08d55034495f%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503107852735217&sdata=94MOnjRlbYikOAFdAsZJM2hIpcPeD7mnmBhtjtX0nlA%3D&reserved=0>:

> @@ -7,8 +7,8 @@

   <connectionStrings>

     <!-- The format of the connection string is "DefaultEndpointsProtocol=https;AccountName=NAME;AccountKey=KEY" -->

     <!-- For local execution, the value can be set either in this config file or through environment variables -->

-    <add name="AzureWebJobsDashboard" connectionString="UseDevelopmentStorage=true"/>

-    <add name="AzureWebJobsStorage" connectionString="UseDevelopmentStorage=true"/>

+    <add name="AzureWebJobsDashboard" connectionString="DefaultEndpointsProtocol=https;AccountName=evitensubminlog;AccountKey=Y+DFscPR1AOYAsPnAjKHJB634P68vNDjSuXkSrOyfEKbSmHEoEA54MxnGktL/I7oJ2/RZULskM4Ttgk91vABzg==;EndpointSuffix=core.windows.net"/>

+    <add name="AzureWebJobsStorage" connectionString="DefaultEndpointsProtocol=https;AccountName=evitensubminlog;AccountKey=Y+DFscPR1AOYAsPnAjKHJB634P68vNDjSuXkSrOyfEKbSmHEoEA54MxnGktL/I7oJ2/RZULskM4Ttgk91vABzg==;EndpointSuffix=core.windows.net"/>

     <add name="DefaultConnection" connectionString="Data Source=MAXIMSH-2010\SQLEXPRESS;Initial Catalog=DataAccess;Integrated Security=True;Connect Timeout=15;Encrypt=False;TrustServerCertificate=True;ApplicationIntent=ReadWrite;MultiSubnetFailover=False" providerName="System.Data.SqlClient"/>

   </connectionStrings>

please don't push such secrets.
As I recommended you, it is better to put all you secrets in key vault. this way you can have you own private environment without the need to replace those secrets.

Also I\m not sure who can see this repo, but you may consider regenerating your storage key now after it became available in GitHub

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23pullrequestreview-86044021&data=02%7C01%7Ceviten%40microsoft.com%7Ccd85e1fe4a20401c2dbf08d55034495f%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503107852735217&sdata=p67BujaWz9ECMJLVyAli0JMe4QRtdCaDlgmfoVzwajU%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FARCVLFx4IEq0n734ajwlHw5p5HDfoVW8ks5tF1l_gaJpZM4RPVhW&data=02%7C01%7Ceviten%40microsoft.com%7Ccd85e1fe4a20401c2dbf08d55034495f%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503107852735217&sdata=6Aygkgcrost%2Flcf8%2BE%2FRfcN%2FOI6iLUb9HaDf%2F2lDoR0%3D&reserved=0>.

bergano65 commented:
agree

From: ischrei [mailto:notifications@github.com]
Sent: Sunday, December 31, 2017 11:50 AM
To: shklar/SubMinimizer <SubMinimizer@noreply.github.com>
Cc: Evgeny Vitenberg <eviten@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [shklar/SubMinimizer] Hot fix (#77)


@ischrei commented on this pull request.

________________________________

In CogsMinimizer/Global.asax.cs<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23discussion_r159136967&data=02%7C01%7Ceviten%40microsoft.com%7Cb3fd9fe7ee2946ce7b0b08d55033d6cd%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503105926785483&sdata=Nk1A4YKq2rptYHlZpHncuAP1mKj82a6iFctXZox1XL4%3D&reserved=0>:

> @@ -21,5 +24,36 @@ protected void Application_Start()

             Database.SetInitializer(new MigrateDatabaseToLatestVersion<DataAccess,

                 CogsMinimizer.Migrations.Configuration>());

         }

+

+

+        protected void Application_BeginRequest(object sender, EventArgs args)

+        {

+            if (Context.Request.Url.Scheme == "https")

+            {

+                return;

+            }

+

+            if (_prevUri != null && _prevUri.AbsoluteUri == Context.Request.Url.AbsoluteUri)

Sorry but I don't understand it. Why do you need to previous URI? what if the previous URI was for another client request?

Anyhow all you need to do is redirect whenever you get a "http" request. I agree you can skip for localhost although it is better to simulate prod in local as much as possible.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23pullrequestreview-86043966&data=02%7C01%7Ceviten%40microsoft.com%7Cb3fd9fe7ee2946ce7b0b08d55033d6cd%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503105926785483&sdata=YAJoNhWgUkjNrRxcJuvWBy7gYV3pBmFZvOJEbVaC2kc%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FARCVLGDf-u6DzFlOnPEnBX0H0HtdWPkaks5tF1i-gaJpZM4RPVhW&data=02%7C01%7Ceviten%40microsoft.com%7Cb3fd9fe7ee2946ce7b0b08d55033d6cd%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503105926785483&sdata=q1O8f4X3VypWlfDCAbCkUCxXWBF3GpLFzPB8SiBqOZc%3D&reserved=0>.

bergano65 commented:
Application locally is running Internet explorer Express server. It reserves some port range for https applications. Port below is in this range. Redirect at localhost doesn’t work under  default port.

From: ischrei [mailto:notifications@github.com]
Sent: Sunday, December 31, 2017 11:46 AM
To: shklar/SubMinimizer <SubMinimizer@noreply.github.com>
Cc: Evgeny Vitenberg <eviten@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [shklar/SubMinimizer] Hot fix (#77)


@ischrei commented on this pull request.

________________________________

In CogsMinimizer/Global.asax.cs<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23discussion_r159136910&data=02%7C01%7Ceviten%40microsoft.com%7Cb9b3ed4ca49b4d8d2c9d08d5503357c3%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503103804188214&sdata=RXjCPaBlUqVh1cX6G2m%2F8IfNULNwiWewEkWOzHli2O8%3D&reserved=0>:

> +            if (_prevUri != null && _prevUri.AbsoluteUri == Context.Request.Url.AbsoluteUri)

+            {

+                return;

+            }

+

+            // This is an insecure connection, so redirect to the secure version

+            UriBuilder uri = new UriBuilder(Context.Request.Url);

+            uri.Scheme = "https";

+

+            if (uri.Host.Equals("localhost"))

+            {

+                uri.Port = 44335;

+            }

+            else

+            {

+                uri.Port = -1;

why? it seems wrong..

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23pullrequestreview-86043909&data=02%7C01%7Ceviten%40microsoft.com%7Cb9b3ed4ca49b4d8d2c9d08d5503357c3%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503103804188214&sdata=P76lymzELgkKYxGKGbLH1zS%2B%2FXu6YWfnZG5PCaiqiSI%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FARCVLJRtd7QWDDGn5ChtUCDBk9ujTh5Hks5tF1fpgaJpZM4RPVhW&data=02%7C01%7Ceviten%40microsoft.com%7Cb9b3ed4ca49b4d8d2c9d08d5503357c3%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503103804188214&sdata=c3qhZxq2ZLjAuxNFrgTrE4Wqmh5vJQp5g03yxNPe01g%3D&reserved=0>.

bergano65 commented:
agree

From: ischrei [mailto:notifications@github.com]
Sent: Sunday, December 31, 2017 11:45 AM
To: shklar/SubMinimizer <SubMinimizer@noreply.github.com>
Cc: Evgeny Vitenberg <eviten@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [shklar/SubMinimizer] Hot fix (#77)


@ischrei commented on this pull request.

________________________________

In CogsMinimizer/Global.asax.cs<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23discussion_r159136892&data=02%7C01%7Ceviten%40microsoft.com%7C4300dffe9aaf4c85a1d608d5503336cb%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503103241767155&sdata=wsqPEsDszFL3fczvmZsq5ZMYjYk56veEOPU4VVwy7LU%3D&reserved=0>:

> @@ -21,5 +24,36 @@ protected void Application_Start()

             Database.SetInitializer(new MigrateDatabaseToLatestVersion<DataAccess,

                 CogsMinimizer.Migrations.Configuration>());

         }

+

+

+        protected void Application_BeginRequest(object sender, EventArgs args)

+        {

+            if (Context.Request.Url.Scheme == "https")

compare with ignore case (not sure if it can also gets "HTTPS" if the client sends it..

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23pullrequestreview-86043887&data=02%7C01%7Ceviten%40microsoft.com%7C4300dffe9aaf4c85a1d608d5503336cb%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503103241767155&sdata=brPdE1ENZIqjLQYw0aMPb3KYe5yE5X6wqgWl%2BrCFtBE%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FARCVLPhh7b6EFIaLuvx6BcszIAywOHjrks5tF1eygaJpZM4RPVhW&data=02%7C01%7Ceviten%40microsoft.com%7C4300dffe9aaf4c85a1d608d5503336cb%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503103241767155&sdata=XVt07MV7SLQmUoInwMNEag3JJcKO9BoIr3r6zTprniY%3D&reserved=0>.

bergano65 commented:
After clean install it will help to run application under debugger. Otherwise you will need manually url (port which will be used for debugger). Would prefer leave it as is.

From: ischrei [mailto:notifications@github.com]
Sent: Sunday, December 31, 2017 11:40 AM
To: shklar/SubMinimizer <SubMinimizer@noreply.github.com>
Cc: Evgeny Vitenberg <eviten@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [shklar/SubMinimizer] Hot fix (#77)


@ischrei commented on this pull request.

________________________________

In CogsMinimizer/CogsMinimizer.csproj<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23discussion_r159136798&data=02%7C01%7Ceviten%40microsoft.com%7C95c27808212e411ac6ca08d550327c4a%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503100123446100&sdata=FH%2FPnS50ShreiG%2BDvaQz%2BWX5F%2Fqn6mrhyViyY2hcSeQ%3D&reserved=0>:

> @@ -18,7 +18,7 @@

     <TargetFrameworkVersion>v4.5.2</TargetFrameworkVersion>

     <MvcBuildViews>false</MvcBuildViews>

     <UseIISExpress>true</UseIISExpress>

-    <IISExpressSSLPort />

+    <IISExpressSSLPort>44335</IISExpressSSLPort>

This is for debugging purposes. If you intended to do it, ok, otherwise please revert

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23pullrequestreview-86043797&data=02%7C01%7Ceviten%40microsoft.com%7C95c27808212e411ac6ca08d550327c4a%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503100123446100&sdata=%2BjQEYEwlxmN12cmAKipc4eOJvX0nkowZGBuvubrso2s%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FARCVLDWrVRLOLV1Y45EKFsdsZpQjJV95ks5tF1Z5gaJpZM4RPVhW&data=02%7C01%7Ceviten%40microsoft.com%7C95c27808212e411ac6ca08d550327c4a%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503100123446100&sdata=pDygSGsTH6tP%2B38y1pxWnfKVv90am5z5B%2FchST4V4D0%3D&reserved=0>.

bergano65 commented:
Please look next iteration

From: ischrei [mailto:notifications@github.com]
Sent: Sunday, December 31, 2017 11:56 AM
To: shklar/SubMinimizer <SubMinimizer@noreply.github.com>
Cc: Evgeny Vitenberg <eviten@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [shklar/SubMinimizer] Hot fix (#77)


@ischrei commented on this pull request.

________________________________

In OfflineSubscriptionManager/Functions.cs<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23discussion_r159137073&data=02%7C01%7Ceviten%40microsoft.com%7Cbf96b7d660104f682f3e08d55034a1e7%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503109344597507&sdata=AX8Qii6KvfgBSJVF7J5tUZ8trsL7lwu7e4X1u3HVAwU%3D&reserved=0>:

> @@ -94,7 +94,11 @@ private static void ReportSubscriptionAnalysisResult(SubscriptionAnalysisResult

             }



             //Add BCC recepients - dev team, as configured in the app config

-            bcc.AddRange(ConfigurationManager.AppSettings["DevTeam"].Split(';').Select(x=> new Email(x)));

+            string devTeam = ConfigurationManager.AppSettings["DevTeam"];

+            if (devTeam != null)

+            {

+                bcc.AddRange(devTeam.Split(';').Select(x => new Email(x)));

add our DL instead

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23pullrequestreview-86044053&data=02%7C01%7Ceviten%40microsoft.com%7Cbf96b7d660104f682f3e08d55034a1e7%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503109344597507&sdata=3ZVWtVNtQUEXU4TaeCj8qrG%2FqwHeZubFRn46WH%2BnSPY%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FARCVLCsPGsn18BnNijGeh9K4vuQZKzleks5tF1oUgaJpZM4RPVhW&data=02%7C01%7Ceviten%40microsoft.com%7Cbf96b7d660104f682f3e08d55034a1e7%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503109344597507&sdata=MT3r9F8%2BETxVSPqhEbQJ7snwaz%2FCK3yQkuDXtyfOUn8%3D&reserved=0>.

ischrei commented on OfflineSubscriptionManager/App.config:
```diff
@@ -7,8 +7,6 @@
-    <add name="AzureWebJobsDashboard" connectionString="UseDevelopmentStorage=true"/>
-    <add name="AzureWebJobsStorage" connectionString="UseDevelopmentStorage=true"/>
     <add name="DefaultConnection" connectionString="Data Source=MAXIMSH-2010\SQLEXPRESS;Initial Catalog=DataAccess;Integrated Security=True;Connect Timeout=15;Encrypt=False;TrustServerCertificate=True;ApplicationIntent=ReadWrite;MultiSubnetFailover=False" providerName="System.Data.SqlClient"/>
   </connectionStrings>
```
I think we will need the properties for local debugging.
feel free to leave those with the connection string to the local dev storage

ischrei commented on OfflineSubscriptionManager/EmailUtils.cs:
```diff
@@ -36,7 +36,7 @@ Subminimizer report
+            string analyzeControllerLink = "https://subminimizer.azurewebsites.net/Subscription/Analyze/";
             string headerLink = HTMLUtilities.CreateHTMLLink($"SubMinimizer report for subscription: {sub.DisplayName}",
                 $"{analyzeControllerLink}/{sub.Id}?OrganizationId={sub.OrganizationId}&DisplayName={sub.DisplayName}");
 
```
you're building the HTML yourself.
Not sure which strings are built into this HTML, but there is a risk in such cases for XSS.
Assume for example you add a message sent by the user, the message can potentially include html tags, even scripts which will be invoked by the browser.
Why don't you use cshtml here?

bergano65 commented:
What way to use cshtml?
I don’t add any external person provided information, mail contains only formatted by me information. What is the risk of using html formatted by me completely?

From: ischrei [mailto:notifications@github.com]
Sent: Monday, January 1, 2018 10:16 AM
To: shklar/SubMinimizer <SubMinimizer@noreply.github.com>
Cc: Evgeny Vitenberg <eviten@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [shklar/SubMinimizer] Hot fix (#77)


@ischrei commented on this pull request.

________________________________

In OfflineSubscriptionManager/EmailUtils.cs<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23discussion_r159151014&data=02%7C01%7Ceviten%40microsoft.com%7C129ccf2df23f47a6448d08d550efd428%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503913341596592&sdata=3fzn2Mmgb8tWD2LZuVvNH2QTWIGunNgx7bEUzel8o94%3D&reserved=0>:

> @@ -36,7 +36,7 @@ Subminimizer report

                             <body>";





-            string analyzeControllerLink = "http://subminimizer.azurewebsites.net/Subscription/Analyze/";

+            string analyzeControllerLink = "https://subminimizer.azurewebsites.net/Subscription/Analyze/";

             string headerLink = HTMLUtilities.CreateHTMLLink($"SubMinimizer report for subscription: {sub.DisplayName}",

                 $"{analyzeControllerLink}/{sub.Id}?OrganizationId={sub.OrganizationId}&DisplayName={sub.DisplayName}");



you're building the HTML yourself.
Not sure which strings are built into this HTML, but there is a risk in such cases for XSS.
Assume for example you add a message sent by the user, the message can potentially include html tags, even scripts which will be invoked by the browser.
Why don't you use cshtml here?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23pullrequestreview-86057755&data=02%7C01%7Ceviten%40microsoft.com%7C129ccf2df23f47a6448d08d550efd428%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503913341596592&sdata=5lZzF19%2FyKxzL2QwPH78jHreR5Mv%2BFFQntE6tYrVwtw%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FARCVLObB6mJKZtNXjrpLbtVhR834cBY-ks5tGJQjgaJpZM4RPVhW&data=02%7C01%7Ceviten%40microsoft.com%7C129ccf2df23f47a6448d08d550efd428%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503913341596592&sdata=l3y5Estc5UU0UmUx6NrwcTAU7jFwWhf7rK49KK2VyC8%3D&reserved=0>.

bergano65 commented:
That’s what fixing now and seems working. Can you check my questions about your corrections? So next iteration will be taking account all our decisions about all corrections.

From: ischrei [mailto:notifications@github.com]
Sent: Monday, January 1, 2018 10:13 AM
To: shklar/SubMinimizer <SubMinimizer@noreply.github.com>
Cc: Evgeny Vitenberg <eviten@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [shklar/SubMinimizer] Hot fix (#77)


@ischrei commented on this pull request.

________________________________

In OfflineSubscriptionManager/App.config<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23discussion_r159150953&data=02%7C01%7Ceviten%40microsoft.com%7C6e6d46bfbd8c4de23ec308d550ef6bae%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503911599869977&sdata=blmxF4rnIQrsy6OerKTCneNJlRlFAtbX7H6p5w2GSPg%3D&reserved=0>:

> @@ -7,8 +7,6 @@

   <connectionStrings>

     <!-- The format of the connection string is "DefaultEndpointsProtocol=https;AccountName=NAME;AccountKey=KEY" -->

     <!-- For local execution, the value can be set either in this config file or through environment variables -->

-    <add name="AzureWebJobsDashboard" connectionString="UseDevelopmentStorage=true"/>

-    <add name="AzureWebJobsStorage" connectionString="UseDevelopmentStorage=true"/>

     <add name="DefaultConnection" connectionString="Data Source=MAXIMSH-2010\SQLEXPRESS;Initial Catalog=DataAccess;Integrated Security=True;Connect Timeout=15;Encrypt=False;TrustServerCertificate=True;ApplicationIntent=ReadWrite;MultiSubnetFailover=False" providerName="System.Data.SqlClient"/>

   </connectionStrings>

I think we will need the properties for local debugging.
feel free to leave those with the connection string to the local dev storage

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23pullrequestreview-86057713&data=02%7C01%7Ceviten%40microsoft.com%7C6e6d46bfbd8c4de23ec308d550ef6bae%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503911599869977&sdata=MmOGJ9lS4U5RE1uRmU5fmg%2FzUwU7ogSN3PvPVB4gpVk%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FARCVLL7tyMGvleMJF7BgfT_CGZNNl8iYks5tGJN1gaJpZM4RPVhW&data=02%7C01%7Ceviten%40microsoft.com%7C6e6d46bfbd8c4de23ec308d550ef6bae%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503911599879991&sdata=fWGJ7RG0p1yZzfk0JAuBbUDRgkkmW1%2BY5L%2FF0B8oyro%3D&reserved=0>.

bergano65 commented:
Again, how to use cshtml?
And is there any risk using html if used only content provided by me?

From: ischrei [mailto:notifications@github.com]
Sent: Monday, January 1, 2018 10:16 AM
To: shklar/SubMinimizer <SubMinimizer@noreply.github.com>
Cc: Evgeny Vitenberg <eviten@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [shklar/SubMinimizer] Hot fix (#77)


@ischrei commented on this pull request.

________________________________

In OfflineSubscriptionManager/EmailUtils.cs<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23discussion_r159151014&data=02%7C01%7Ceviten%40microsoft.com%7C129ccf2df23f47a6448d08d550efd428%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503913341596592&sdata=3fzn2Mmgb8tWD2LZuVvNH2QTWIGunNgx7bEUzel8o94%3D&reserved=0>:

> @@ -36,7 +36,7 @@ Subminimizer report

                             <body>";





-            string analyzeControllerLink = "http://subminimizer.azurewebsites.net/Subscription/Analyze/";

+            string analyzeControllerLink = "https://subminimizer.azurewebsites.net/Subscription/Analyze/";

             string headerLink = HTMLUtilities.CreateHTMLLink($"SubMinimizer report for subscription: {sub.DisplayName}",

                 $"{analyzeControllerLink}/{sub.Id}?OrganizationId={sub.OrganizationId}&DisplayName={sub.DisplayName}");



you're building the HTML yourself.
Not sure which strings are built into this HTML, but there is a risk in such cases for XSS.
Assume for example you add a message sent by the user, the message can potentially include html tags, even scripts which will be invoked by the browser.
Why don't you use cshtml here?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23pullrequestreview-86057755&data=02%7C01%7Ceviten%40microsoft.com%7C129ccf2df23f47a6448d08d550efd428%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503913341596592&sdata=5lZzF19%2FyKxzL2QwPH78jHreR5Mv%2BFFQntE6tYrVwtw%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FARCVLObB6mJKZtNXjrpLbtVhR834cBY-ks5tGJQjgaJpZM4RPVhW&data=02%7C01%7Ceviten%40microsoft.com%7C129ccf2df23f47a6448d08d550efd428%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503913341596592&sdata=l3y5Estc5UU0UmUx6NrwcTAU7jFwWhf7rK49KK2VyC8%3D&reserved=0>.

bergano65 commented:
Sorry, ping?

From: Evgeny Vitenberg
Sent: Monday, January 1, 2018 9:14 PM
To: 'shklar/SubMinimizer' <reply@reply.github.com>; shklar/SubMinimizer <SubMinimizer@noreply.github.com>
Cc: Author <author@noreply.github.com>
Subject: RE: [shklar/SubMinimizer] Hot fix (#77)

What way to use cshtml?
I don’t add any external person provided information, mail contains only formatted by me information. What is the risk of using html formatted by me completely?

From: ischrei [mailto:notifications@github.com]
Sent: Monday, January 1, 2018 10:16 AM
To: shklar/SubMinimizer <SubMinimizer@noreply.github.com<mailto:SubMinimizer@noreply.github.com>>
Cc: Evgeny Vitenberg <eviten@microsoft.com<mailto:eviten@microsoft.com>>; Author <author@noreply.github.com<mailto:author@noreply.github.com>>
Subject: Re: [shklar/SubMinimizer] Hot fix (#77)


@ischrei commented on this pull request.

________________________________

In OfflineSubscriptionManager/EmailUtils.cs<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23discussion_r159151014&data=02%7C01%7Ceviten%40microsoft.com%7C129ccf2df23f47a6448d08d550efd428%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503913341596592&sdata=3fzn2Mmgb8tWD2LZuVvNH2QTWIGunNgx7bEUzel8o94%3D&reserved=0>:

> @@ -36,7 +36,7 @@ Subminimizer report

                             <body>";





-            string analyzeControllerLink = "http://subminimizer.azurewebsites.net/Subscription/Analyze/";

+            string analyzeControllerLink = "https://subminimizer.azurewebsites.net/Subscription/Analyze/";

             string headerLink = HTMLUtilities.CreateHTMLLink($"SubMinimizer report for subscription: {sub.DisplayName}",

                 $"{analyzeControllerLink}/{sub.Id}?OrganizationId={sub.OrganizationId}&DisplayName={sub.DisplayName}");



you're building the HTML yourself.
Not sure which strings are built into this HTML, but there is a risk in such cases for XSS.
Assume for example you add a message sent by the user, the message can potentially include html tags, even scripts which will be invoked by the browser.
Why don't you use cshtml here?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshklar%2FSubMinimizer%2Fpull%2F77%23pullrequestreview-86057755&data=02%7C01%7Ceviten%40microsoft.com%7C129ccf2df23f47a6448d08d550efd428%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503913341596592&sdata=5lZzF19%2FyKxzL2QwPH78jHreR5Mv%2BFFQntE6tYrVwtw%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FARCVLObB6mJKZtNXjrpLbtVhR834cBY-ks5tGJQjgaJpZM4RPVhW&data=02%7C01%7Ceviten%40microsoft.com%7C129ccf2df23f47a6448d08d550efd428%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636503913341596592&sdata=l3y5Estc5UU0UmUx6NrwcTAU7jFwWhf7rK49KK2VyC8%3D&reserved=0>.
